packages/building-blocks/src/dropdown/index.tsx

```diff
@@ -11,39 +11,44 @@ import { getNextSibling, getPreviousSibling } from "../utils/dom-query";
 import { DropdownButton, DropdownMenu, DropdownWrapper } from "./styled";
 
 export interface Props extends HTMLAttributes<HTMLElement> {
+  alignMenuLeft?: boolean;
   buttonStyle?: React.CSSProperties;
-  isList?: boolean;
   label?: string;
   listLabel?: string;
   text?: JSX.Element | string;
   nav?: boolean;
-  AlignMenuLeft?: boolean;
 }
 
 /**
  * Renders a dropdown menu. By default, only a passed "text" is rendered. If clicked,
  * a floating div is rendered below the "text" with list contents inside. Clicking anywhere
  * outside the floating div will close the dropdown.
  */
-export const Dropdown = ({
+const Dropdown = ({
+  alignMenuLeft,
   children,
   className,
   id,
-  isList,
   label,
   listLabel,
   text,
-  AlignMenuLeft,
   buttonStyle
 }: Props): JSX.Element => {
   const [open, setOpen] = useState(false);
-  const containerRef = useRef<HTMLLIElement>(null);
+
+  const containerRef = useRef<HTMLSpanElement>(null);
 
   const toggleOpen = useCallback(() => setOpen(!open), [open, setOpen]);
 
   // Argument for document.querySelectorAll to target focusable elements.
   const queryId = `#${id} button, #${id}-label`;
 
+  const isList = Array.isArray(children)
```

Comment:
Nice!

Reply:
Cool to see `isValidElement` in use. I'm guessing this was needed because my example didn't have typescript?

Reply:
yeah I was getting weird type errors. perfect use of `isValidElement` here!